cloud/amazon/ec2_vpc_dhcp_options.py

```diff
@@ -128,105 +159,207 @@
+        netbios_name_servers=dict(type='list', default=None),
+        netbios_node_type=dict(type='int', default=None),
+        vpc_id=dict(type='str', default=None),
         delete_old=dict(type='bool', default=True),
-        inherit_existing=dict(type='bool', default=False)
+        inherit_existing=dict(type='bool', default=False),
+        tags=dict(type='dict', default=None, aliases=['resource_tags']),
+        state=dict(type='str', default='present', choices=['present', 'absent'])
         )
     )
 
     module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=True)
     params = module.params
+    found = False
+    changed = False
+    new_options = collections.defaultdict(lambda: None)
 
   
     region, ec2_url, boto_params = get_aws_connection_info(module)
     connection = connect_to_aws(boto.vpc, region, **boto_params)
-  
-    inherit_existing = params['inherit_existing']
-  
-    existing_options = _get_associated_dhcp_options(params['vpc_id'], connection)
-    new_options = collections.defaultdict(lambda: None)
-  
-    new_options['domain-name-servers'] = _get_updated_option( params['dns_servers'],
-            existing_options.options.get('domain-name-servers'), inherit_existing)
-  
-    new_options['netbios-name-servers'] = _get_updated_option(params['netbios_name_servers'],
-            existing_options.options.get('netbios-name-servers'), inherit_existing)
-  
-  
-    new_options['ntp-servers'] = _get_updated_option(params['ntp_servers'],
-            existing_options.options.get('ntp-servers'), inherit_existing)
-  
-    # HACK: Why do I make the next two lists? The boto api returns a list if present, so
-    # I need this to properly compare so == works.
-  
-    # HACK: netbios-node-type is an int, but boto returns a string. So, asking for an int from Ansible
-    # for data validation, but still need to cast it to a string
-    new_options['netbios-node-type'] = _get_updated_option(
-            [str(params['netbios_node_type'])], existing_options.options.get('netbios-node-type'),
-            inherit_existing)
-  
-    new_options['domain-name'] = _get_updated_option(
-            [params['domain_name']], existing_options.options.get('domain-name'), inherit_existing)
-  
-    if existing_options and new_options == existing_options.options:
-        module.exit_json(changed=False)
-  
-    if new_options['netbios-node-type']:
-        new_options['netbios-node-type'] = new_options['netbios-node-type'][0]
-  
-    if new_options['domain-name']:
-        new_options['domain-name'] = new_options['domain-name'][0]
-  
-    if not module.check_mode:
-        dhcp_option = connection.create_dhcp_options(new_options['domain-name'],
-                new_options['domain-name-servers'], new_options['ntp-servers'],
-                new_options['netbios-name-servers'], new_options['netbios-node-type'])
+
+    # First check if we were given a dhcp_options_id
+    if not params['dhcp_options_id']:
```

Comment:
In general I agree... I have a sense that it may take significant refactoring to achieve that though
(because there's a bunch of things we *only* do if dhcp_options_id wasn't provided, and they happen first)